src/mc.rs

```diff
@@ -1942,18 +1943,18 @@ unsafe extern "C" fn resize_c_erased<BD: BitDepth>(
     dx: i32,
     mx0: i32,
     bitdepth_max: i32,
-    src: *const FFISafe<Rav1dPictureDataComponentOffset>,
-    dst: *const FFISafe<WithOffset<PicOrBuf<AlignedVec64<u8>>>>,
+    src: FFISafeRav1dPictureDataComponentOffset,
+    dst: WithOffset<*const FFISafe<PicOrBuf<AlignedVec64<u8>>>>,
 ) {
-    // SAFETY: Was passed as `FFISafe::new(_)` in `resize::Fn::call`.
-    let dst = *unsafe { FFISafe::get(dst) };
-    // SAFETY: Was passed as `FFISafe::new(_)` in `resize::Fn::call`.
-    let src = *unsafe { FFISafe::get(src) };
+    // SAFETY: Was passed as `WithOffset::into_ffi_safe(_)` in `resize::Fn::call`.
+    let dst = unsafe { FFISafe::from_with_offset(dst) };
+    // SAFETY: Was passed as `WithOffset::into_ffi_safe(_)` in `resize::Fn::call`.
+    let src = unsafe { FFISafe::from_with_offset(src) };
     let dst_w = dst_w as usize;
     let h = h as usize;
     let src_w = src_w as usize;
     let bd = BD::from_c(bitdepth_max);
-    resize_rust(dst, src, dst_w, h, src_w, dx, mx0, bd)
+    resize_rust(dst.map(|data| *data), src, dst_w, h, src_w, dx, mx0, bd)
```

Comment:
Could you put this in a separate `let dst = dst.map(|data| *data);` line?  I've been limiting these calls to variables just for consistency in these `fn *_c_erased` calling `fn *_rust` functions.